pystove/pystove.py

```diff
@@ -382,11 +384,13 @@ async def get_name_and_ip():
             """Get stove name and IP."""
             stove_id = await self._get_json('http://' + self.stove_host
                                             + STOVE_ID_URL)
-            if None in [stove_id.get(DATA_NAME), stove_id.get(DATA_IP)]:
-                _LOGGER.warning("Unable to read stove name and/or IP.")
+            if None in [stove_id.get(DATA_NAME), stove_id.get(DATA_IP), stove_id[DATA_MDNS]]:
```

Comment:
`DATA_MDNS` is not guaranteed to be in the response. Use `stove_id.get(DATA_MDNS)` instead.

Also, is there a minimum stove firmware version where this value is present? If so, we should handle the case where mdns is not available but the other info is.

Reply:
> is there a minimum stove firmware version where this value is present

I don't have such insights. My stoves airbox reports the following sw versions:
firmware_version: 3.23.0
algorithm: DS.DSB2.221129
wifi: 12.6.0
remote: 3.6.0

Reply:
To follow-up on this, as I now have access to a stove again, I see the following values:

firmware 3.23.0
algorithm HW.3600.221202
wifi 12.5.0
remote 1.12.0

This version does not provide mdns information in `/esp/get_stove_data`.

Reply:
I assume the algorithm has nothing to do with it and the remote too. Firmware is equal to mine so looks like the Wi-Fi is the sw part that implements the REST interface. 

Question is if there is any other unique identification. 

BTW where did you get all the infos about the API? Sniffing?